R/diagnosis_helper_functions.R

```diff
@@ -85,3 +85,86 @@ print.summary.diagnosis <- function(x, ...) {
   cat("\n")
   invisible(x)
 }
+
+
+
+#' Clean up DeclareDesign diagnosis object for printing
+#'
+#' If diagnosands are bootstrapped, se's are put in parenthese on a second line and rounded to \code{digits}.
+#' Function uses presence of "se(" to identify bootrapped diagnoses; avoid errors by not using "se(" in naming of diagnosands.
+#'
+#' @param diagnosis An object from \code{declare_design}
+#' @param digits Number of digits.
+#' @param col.names Allows user to provide names of columns for output. If NULL uses names from diagnosis object, if "default" uses names of default diagnosands.
+#' @param n_text_fields Number of initial text fields in diagnosis that do not have associated standard errors.
+#' @return A formatted text table with bootstrapped standard errors in parentheses.
+#' @export
+#'
+#' @examples
+#' my_population <- declare_population(N = 50, noise = rnorm(N))
+#'
+#' my_potential_outcomes <-
+#'   declare_potential_outcomes(Y_Z_0 = noise,
+#'                              Y_Z_1 = noise + rnorm(N, mean = 2, sd = 2))
+#'
+#' my_assignment <- declare_assignment(m = 25)
+#'
+#' pate <- declare_estimand(mean(Y_Z_1 - Y_Z_0), label = "pate")
+#'
+#' pate_estimator <- declare_estimator(Y ~ Z, estimand = pate, label = "test")
+#'
+#'
+#' my_design <- declare_design(my_population,
+#'                             my_potential_outcomes,
+#'                             pate,
+#'                             my_assignment,
+#'                             pate_estimator)
+#'
+#' my_diagnosis <- diagnose_design(my_design)
+#'
+#' reshape_diagnosis(my_diagnosis)
+#'
+#'
+reshape_diagnosis <- function(diagnosis, digits = 2){
+
+  diagnosands_df <- diagnosis$diagnosands
+  diagnosands_df_names <- names(diagnosands_df)
+
+  group_names <- diagnosands_df_names[1:3]
+  value_names <- diagnosands_df_names[-(1:3)]
+
+
+  diagnosands_only_names <- value_names[which(substr(value_names, start = 1, stop = 3) != "se(")]
```

Comment:
`grep('^se(', value_names, value=TRUE)`

Reply:
I think I'm not quite grokking this grep. I get:

```
grep('^se(', value_names, value=TRUE)
Error in grep("^se(", value_names, value = TRUE) : 
  invalid regular expression '^se(', reason 'Missing ')''
```

Reply:
got it `grep('^se\\(', value_names, value=TRUE)`